### PR TITLE
Add localFocalPoint to ScaleDetector

### LINKED ISSUE #33818: ScaleGestureRecognizer isn't transforming PointerEvents to the local coordinate system of the event receiver
After pull [32192](https://github.com/flutter/flutter/pull/32192), pointer events should be local to the event receiver. But I think I'm still seeing the previous behavior, of global PointerEvents.

I'm still getting global events from GestureDetector.onScaleUpdate. I created a minimal example, where I've got a box that renders the initial focal point of the gesture, as well as the current focal point: [example on gist.github](https://gist.github.com/mtcliatt/24fc63d7d4f58f1e928be1da47c90b2f).

Here's a screenshot of the example that shows the focal points below the box that contains the gesture detector: [link](https://drive.google.com/open?id=1vse_a0u9nISnlPSpLQDPJgDinv6SvNer).

Is this expected?

I'm running:
Flutter 1.6.8-pre.43 • channel master • https://github.com/flutter/flutter.git
Framework • revision cd1d40a (7 hours ago) • 2019-06-03 11:08:30 -0700
Engine • revision f9ce01674e
Tools • Dart 2.3.2 (build 2.3.2-dev.0.0 224f82c21c)

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

Adds the localFocalPoint to ScaleStartDetails and ScaleUpdateDetails. I forgot to do this in https://github.com/flutter/flutter/pull/32192.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/33818

## Tests

I added the following tests:

* Verify that details contain local focalPoint.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/gestures/scale.dart [hunk 1]
```diff
@@ -34,15 +36,31 @@ class ScaleStartDetails {
...
+  ScaleStartDetails({ this.focalPoint = Offset.zero, Offset localFocalPoint, })
+    : assert(focalPoint != null), localFocalPoint = localFocalPoint ?? focalPoint;
 
   /// The initial focal point of the pointers in contact with the screen.
+  ///
   /// Reported in global coordinates.
+  ///
+  /// See also:
+  ///
+  ///  * [localFocalPoint], which is the same value reported in global
```
Comment (mtcliatt) [inline packages/flutter/lib/src/gestures/scale.dart]: "local" (and on 93)

Sorry for the drive-by review, I was just looking through the commit to learn more and spotted this easy-to-miss slip.

Comment (justinmc) [reply packages/flutter/lib/src/gestures/scale.dart]: +1

**Review by justinmc — APPROVED** [review]
LGTM other than the quick local/global swap mentioned by @mtcliatt.

Bonus points for adding a parameter that rhymes.
